doc/c-style-guide.md

```diff
@@ -72,8 +72,16 @@ Symbol Names
 
 - Non-static functions must be prefixed by `grpc_`
 - Static functions must *not* be prefixed by `grpc_`
+- Typenames of `struct`s , `union`s, and `enum`s must be prefixed by `grpc_` if
+  they are declared in a header file. They must not be prefixed by `grpc_` if
+  they are declared in a source file.
 - Enumeration values and `#define` names must be uppercase. All other values
   must be lowercase.
+- Enumeration values or `#define` names defined in a header file must be
+  prefixed with `GRPC_` (except for `#define` macros that are being used to
```

Comment:
I don't think this exception is needed. I can't imagine a scenario where we use macro substitutions written in global headers for static functions

Reply:
Ah, I see the issue. I disagree. We use #defines to substitute for functions, so they should have names that start with `grpc_` when appropriate, not `GRPC_`. Of course, we also have #defines that do more than just the function (like `GRPC_ERROR_CREATE`) so that makes sense to capitalize. I wasn't talking about #defines in headers to substitute static functions.

Reply:
Ah yes! You are right.